### PR TITLE
[feat]: [SPG-777]: Adds intro documentation to API spec

### PR DESCRIPTION
**Pull request by mankrit-singh**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41654)

### CODE REVIEW

File: 970-ng-commons/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -6,9 +6,72 @@ info:
   version: '1.0'
   description: |
     This is the Open Api Spec 3 for Harness. This is under active development. Beware of the breaking change with respect to the generated code stub.
```
[the remark this answers is not in the document]
Comment (mankrit-singh) [reply 970-ng-commons/contracts/openapi/v1/openapi.yaml:8]: yes this works, and a process for handling breaking changes is underway

File: 970-ng-commons/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -6,9 +6,72 @@ info:
...
+    To authenticate with the Harness API, you need to:
+      1. Generate an API token on the Harness Platform.
+      2. Send the API token you generate in the `x-api-key field` in each request.
+    
+    ### Generate an API token
+    
+    To generate an API token, complete the following steps:
+      1. Go to the [Harness Platform](app.harness.io).
+      2. On the left-hand navigation, click **My Profile**.
+      3. Click **+API Key**, enter a name for your key, then click **Save**.
```
[the remark this answers is not in the document]
Comment (mankrit-singh) [reply 970-ng-commons/contracts/openapi/v1/openapi.yaml:43]: ack

File: 970-ng-commons/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -6,9 +6,72 @@ info:
...
+      4. Within the API Key tile, click **+Token**.
+      5. Enter a name for your token and click **Generate Token**.
+    **Important**: Make sure to save your token securely, you can't see it again after leaving the page.
```
[the remark this answers is not in the document]
Comment (mankrit-singh) [reply 970-ng-commons/contracts/openapi/v1/openapi.yaml:46]: using the second one

**Review by deepakbolangady — APPROVED** [review]
LGTM! Thanks for getting this content in and then reviewed as well.

**Review by deepakbolangady — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41654)* status: 4. unresolved discussions.

**Review by mankrit-singh — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41654)* status: 9. unresolved discussions. 

___
*[`970-ng-commons/contracts/openapi/v1/openapi.yaml` line 50 at r1](https://reviewable.io/reviews/harness/harness-core/41654#-NJOKTzzGqX2ckjCs5rX-r1-50:-NJPiRbC0cusZcd7yeW4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8cf7b986a8e94564d0bc527d67466f5aa8003bbf/970-ng-commons/contracts/openapi/v1/openapi.yaml#L50)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

Terminate the sentence after "parameter". 
</blockquote></details>

Done.
___
*[`970-ng-commons/contracts/openapi/v1/openapi.yaml` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/41654#-NJOKTzzGqX2ckjCs5rX-r1-55:-NJPiRpf-8lYV7h23tVD:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8cf7b986a8e94564d0bc527d67466f5aa8003bbf/970-ng-commons/contracts/openapi/v1/openapi.yaml#L55)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

Insert a comma after "response". Better still, terminate the sentence at that point. 
Replace "to test these out" with "test the examples".
</blockquote></details>

Done.
___
*[`970-ng-commons/contracts/openapi/v1/openapi.yaml` line 61 at r1](https://reviewable.io/reviews/harness/harness-core/41654#-NJOKTzzGqX2ckjCs5rX-r1-61:-NJPiS9ECcIaIR6U7_LU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8cf7b986a8e94564d0bc527d67466f5aa8003bbf/970-ng-commons/contracts/openapi/v1/openapi.yaml#L61)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

The version number represents the core API and does not change frequently. The version number changes only if there is a significant departure from the basic underpinnings of the existing API. For example, the version number would change with a system-wide refactoring of core concepts or resources.
</blockquote></details>

Done.
___
*[`970-ng-commons/contracts/openapi/v1/openapi.yaml` line 62 at r1](https://reviewable.io/reviews/harness/harness-core/41654#-NJOKTzzGqX2ckjCs5rX-r1-62:-NJPiSLBBnNMJKdsLONh:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8cf7b986a8e94564d0bc527d67466f5aa8003bbf/970-ng-commons/contracts/openapi/v1/openapi.yaml#L62)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

1) "The version number is represented in the API path." (Terminate the sentence after "API path".)
2) "For example, the v1 in `api.harness.io/v1/ is the version number.' (prefer complete sentences where possible or enclose current sentence in parentheses, as part of the previous sentence)
</blockquote></details>

Done.
___
*[`970-ng-commons/contracts/openapi/v1/openapi.yaml` line 66 at r1](https://reviewable.io/reviews/harness/harness-core/41654#-NJOKTzzGqX2ckjCs5rX-r1-66:-NJPiShHA2IbvJnecuv8:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8cf7b986a8e94564d0bc527d67466f5aa8003bbf/970-ng-commons/contracts/openapi/v1/openapi.yaml#L66)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

Also, "The limit defaults to 30. Its maximum value is 100." (Works?)
</blockquote></details>

Done.
___
*[`970-ng-commons/contracts/openapi/v1/openapi.yaml` line 68 at r1](https://reviewable.io/reviews/harness/harness-core/41654#-NJOKTzzGqX2ckjCs5rX-r1-68:-NJPiT13CD7-i7dSo9du:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8cf7b986a8e94564d0bc527d67466f5aa8003bbf/970-ng-commons/contracts/openapi/v1/openapi.yaml#L68)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

Can we clarify whether we mean that:
* Link headers are applied to both metadata AND related links?
Or do we mean that
* We use link headers for metadata, and we also use related links. 
I can propose a correction only after I understand this statement better. The "which" clause will also have to change based on that understanding.
</blockquote></details>

Done.

**Review by deepakbolangady — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41654)* status: :shipit: complete! ✓

**Pull request merged**